python/pyspark/ml/feature.py

```diff
@@ -5313,7 +5313,7 @@ class VectorAssembler(
     +---+---+----+-------------+
     |  a|  b|   c|     features|
     +---+---+----+-------------+
-    |1.0|2.0|NULL|[1.0,2.0,NaN]|
+    |1.0|2.0|null|[1.0,2.0,NaN]|
```

Comment:
This is another difference from the existing behavior, @Yikf .

Reply:
Existing behavior should be `null`, `NULL` is changed in https://github.com/apache/spark/pull/40699, I think we should do the same with map type, default to `null` instead of `NULL`

Reply:
Thank you for the pointer. I'm not sure I can agree with #40699 there. I'm tracking the comment on that PR.